### PR TITLE
11 delete comments

### PR DESCRIPTION
**Pull request by TomReece1**
Added users and the ability to log in and change user so that users can only delete their own comments and make comments if they're logged in etc.

### CODE REVIEW

**Comment by P-Copley** [comment]
UX wise when I change user I don't get any feedback to show me that it's happened. Consider changing the page or some other feedback to let the user know the change has worked.

File: src/components/ChangeUser.jsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+function ChangeUser() {
+  const [users, setUsers] = useState([]);
+  //New state that is all possible user profiles
+  const [isLoading, setIsLoading] = useState(true);
+  const { user, setUser } = useContext(UserContext);
+  //Grab and use context and destructure the setUser function part only
+  const usersCopy = [...users];
+
+  function displayUsers(username) {
```
Comment (P-Copley) [inline src/components/ChangeUser.jsx]:
```js
  function displayUsers(username) {
```
Don't think you ever use this function ^ Go ahead and remove it

File: src/components/ChangeUser.jsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  return (
+    <section>
+      <h2>Choose a user:</h2>
+      {isLoading ? (
+        <p>Loading...</p>
+      ) : (
+        <ul>
+          {users.map((profile, index) => {
+            const isMe = profile.username === user.username;
```
Comment (P-Copley) [inline src/components/ChangeUser.jsx]:
```js
            const isMe = profile.username === user.username;
```
Not sure if you were planning on using this to do something different with the current logged in user?

**Comment by TomReece1** [comment]
> UX wise when I change user I don't get any feedback to show me that it's happened. Consider changing the page or some other feedback to let the user know the change has worked.

Agree, I'll send them straight back to the homepage where it will say Welcome username.

File: src/components/CommentList.jsx [hunk 3]
```diff
@@ -52,11 +56,18 @@ function CommentList() {
       {isOpen && (
         <div>
           <h2>Comments</h2>
+          {commentDeleted && <p>Comment was delted</p>}
```
Comment (P-Copley) [inline src/components/CommentList.jsx]: `delted` is misspelled

File: src/components/ChangeUser.jsx (hunk 2 above)
Comment (TomReece1) [reply src/components/ChangeUser.jsx]: Ended up not being useful, will delete

File: src/components/CommentList.jsx (hunk 3 above)
Comment (TomReece1) [reply src/components/CommentList.jsx]: > `delted` is misspelled

corrected

**Comment by P-Copley** [comment]
Hey @TomReece1 

Take a look at the above comments but the functionality looks great overall. Go ahead and merge when you're ready 😁

File: src/components/ChangeUser.jsx (hunk 1 above)
Comment (TomReece1) [reply src/components/ChangeUser.jsx]: removed

**Comment by TomReece1** [comment]
> > UX wise when I change user I don't get any feedback to show me that it's happened. Consider changing the page or some other feedback to let the user know the change has worked.
> 
> Agree, I'll send them straight back to the homepage where it will say Welcome username.

Done

**Pull request merged**